loadtest/store/memstore/store.go

```diff
@@ -176,16 +176,24 @@ func (s *MemStore) SetTeams(teams []*model.Team) error {
 	return nil
 }
 
-func (s *MemStore) SetChannelMembers(channelId string, channelMembers *model.ChannelMembers) error {
+// SetChannelMembers stores the given channel members in the store.
+func (s *MemStore) SetChannelMembers(channelMembers *model.ChannelMembers) error {
 	if channelMembers == nil {
 		return errors.New("channelMembers should not be nil")
 	}
-	membersMap := make(map[string]*model.ChannelMember)
-	members := *channelMembers
-	for _, m := range members {
-		membersMap[m.UserId] = &m
+
+	for _, member := range *channelMembers {
+		// Initialize the maps as necessary.
+		if s.channelMembers == nil {
+			s.channelMembers = make(map[string]map[string]*model.ChannelMember)
+		}
+		if s.channelMembers[member.ChannelId] == nil {
+			s.channelMembers[member.ChannelId] = make(map[string]*model.ChannelMember)
+		}
```

Comment:
Just pointing out that we are *slightly* changing the behaviour with these changes.

Before we would replace this map whether or not it was nil. That was done to keep the state consistent since we currently don't keep it updated through WebSocket and at each `SimpleController.reload()` we expect to get the store loaded properly.

That said, this is a recurring problem, especially when we need to store partial results and we cannot just replace what we have so I am thinking a good option might be to expose some functionality to clear the store (or part of it) and use that when needed such as in the `reload()` case.

Reply:
Right. I can always do a `s.channelMembers = make(map[string]map[string]*model.ChannelMember)` unconditionally to reset the map. But I agree that we need partial updates also, as that is more close to how the client will behave.

Keeping this as-is for now.